src/main/java/org/apache/sling/commons/threads/impl/ThreadPoolMetricsGauges.java

```diff
@@ -0,0 +1,107 @@
+
+class ThreadPoolMetricsGauges {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ThreadPoolMetricsGauges.class);
+
+    // not exposing MaxThreadAge as bean.getMaxThreadAge() is deprecated due
+    // to SLING-6261 and always returns -1
+    private static final List<String> IGNORED_ATTRIBUTES = Collections.singletonList("MaxThreadAge");
+
+    private static final Map<Class<?>, Class<?>> WRAPPER_TO_PRIMITIVE = new HashMap<>();
+
+    static {
+        WRAPPER_TO_PRIMITIVE.put(Integer.class, int.class);
+        WRAPPER_TO_PRIMITIVE.put(Long.class, long.class);
+        WRAPPER_TO_PRIMITIVE.put(Boolean.class, boolean.class);
+    }
+
+    private static final Map<Class<?>, Object> CLASS_TO_DEFAULT_VALUE = new HashMap<>();
+
+    static {
+        CLASS_TO_DEFAULT_VALUE.put(Integer.class, -1);
+        CLASS_TO_DEFAULT_VALUE.put(Long.class, -1L);
+        CLASS_TO_DEFAULT_VALUE.put(Boolean.class, false);
+        CLASS_TO_DEFAULT_VALUE.put(String.class, "");
+    }
+
+    private ThreadPoolMetricsGauges() {
+    }
+
+    private static boolean isTypeOfClassOrPrimitive(MBeanAttributeInfo attr, Class<?> clazz) {
+        if (clazz.getName().equals(attr.getType())) {
+            return true;
+        } else {
+            Class<?> primitive = WRAPPER_TO_PRIMITIVE.get(clazz);
+            return primitive != null && primitive.getName().equals(attr.getType());
+        }
+    }
+
+    private static <T> Gauge<T> createGauge(final DynamicMBean bean, final String name, final Object defaultValue) {
+        return new Gauge<T>() {
+            @SuppressWarnings("unchecked")
+            public T getValue() {
+                try {
+                    return (T) bean.getAttribute(name);
+                } catch (Exception e) {
+                    LOGGER.warn("cannot obtain MBean attribute named " + name, e);
+                    return (T) defaultValue;
+                }
+            }
+        };
+    }
+
+    public static Map<String, Gauge<?>> create(final DynamicMBean bean) {
+        Map<String, Gauge<?>> gauges = new HashMap<>();
+        for (MBeanAttributeInfo attr : bean.getMBeanInfo().getAttributes()) {
+            String name = attr.getName();
+            if (IGNORED_ATTRIBUTES.contains(name)) {
+                LOGGER.debug("ignoring MBean attribute {}", name);
+            } else {
+                boolean gaugeForAttribueCreated = false;
+
+                final Set<Class<?>> supportedClasses = CLASS_TO_DEFAULT_VALUE.keySet();
+                for (Class<?> clazz : supportedClasses) {
+                    if (isTypeOfClassOrPrimitive(attr, clazz)) {
+                        gauges.put(name, createGauge(bean, name, CLASS_TO_DEFAULT_VALUE.get(clazz)));
+                        gaugeForAttribueCreated = true;
+                    }
+                }
+
+                if (!gaugeForAttribueCreated) {
+                    LOGGER.warn("no gauge for attribute {} of type {} created", name, attr.getType());
```

Comment:
It would be good to say _why_ it was not created.

Reply:
@rombert Fixed with https://github.com/apache/sling-org-apache-sling-commons-threads/pull/1/commits/dbf39320c4ba1ff29c22166a70fb2ce5fecae3a3